pkg/resource/introspect.go

```diff
@@ -0,0 +1,98 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package resource
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	GetNodeResourcesPath = "/node/"
+	GetAllResourcesPath  = "/resources"
+)
+
+type IntrospectHandler struct {
+	Log             logr.Logger
+	BindAddress     string
+	ResourceManager ResourceManager
+}
+
+// StartENICleaner starts the ENI Cleaner routine that cleans up dangling ENIs created by the controller
```

Comment:
Is this comment on `StartENICleaner` correct here? I couldn't directly see the association.

Reply:
Good Catch, I will fix this in follow up PR!